src/qzone3tg/app/interact.py

```diff
@@ -77,20 +78,31 @@ async def reply_markup(self, feed: FeedContent):
         return interact_tasker_hook
 
     @property
-    def _qr_hook_cls(self):
-        class interact_qr_hook(super()._qr_hook_cls):
+    def _login_hook_cls(self):
+        cls = super()._login_hook_cls
+
+        class interact_qr_hook(cls._get_base(LoginMethod.qr)):
             def qr_markup(self):
                 btnrefresh = InlineKeyboardButton("刷新", callback_data="qr:refresh")
                 btncancel = InlineKeyboardButton("取消", callback_data="qr:cancel")
                 return InlineKeyboardMarkup([[btnrefresh, btncancel]])
 
-        return interact_qr_hook
+        class interact_login_hook(cls):
+            @classmethod
+            def _get_base(cls, meth: LoginMethod):
+                if meth == LoginMethod.qr:
+                    return interact_qr_hook
+                return super()._get_base(meth)
+
+        return interact_login_hook
 
     @property
     def _feed_hook_cls(self):
-        class interact_feed_hook(super()._feed_hook_cls):
-            async def HeartbeatRefresh(hook, num: int):  # type: ignore
-                await DefaultFeedHook.HeartbeatRefresh(hook, num)
+        cls = super()._feed_hook_cls
+
+        class interact_feed_hook(cls):
+            async def HeartbeatRefresh(_self, num: int):
+                await super().HeartbeatRefresh(num)
```

Comment:
Super is BaseApp._feed_hook_cls, which contains `fetch`. So fetch here is unnecessary and caused duplicated feed message.